lib/matplotlib/_afm.py

```diff
@@ -163,22 +176,29 @@ def _parse_header(fh):
     return d
 
 
-CharMetrics = namedtuple('CharMetrics', 'width, name, bbox')
-CharMetrics.__doc__ = """
+class CharMetrics(NamedTuple):
+    """
     Represents the character metrics of a single character.
 
     Notes
     -----
     The fields do currently only describe a subset of character metrics
     information defined in the AFM standard.
     """
+
+    width: float
+    name: str
+    bbox: tuple[int, int, int, int]
+
+
 CharMetrics.width.__doc__ = """The character width (WX)."""
 CharMetrics.name.__doc__ = """The character name (N)."""
 CharMetrics.bbox.__doc__ = """
     The bbox of the character (B) as a tuple (*llx*, *lly*, *urx*, *ury*)."""
```

Comment:
```suggestion

    width: float
    """The character width (WX)."""
    name: str
    """The character name (N)."""
    bbox: tuple[int, int, int, int]
    """The bbox of the character (B) as a tuple (*llx*, *lly*, *urx*, *ury*)."""
```

Reply:
Unfortunately, this does not seem to work:
```
>>> from matplotlib import _afm
>>> _afm.CharMetrics.width
_tuplegetter(0, 'Alias for field number 0')
```